index.js

```diff
@@ -81,11 +94,14 @@ class MMST extends EventEmitter {
     this.connectedPeers.add(stringId)
     connection.once('close', () => {
       this.connectedPeers.delete(stringId)
+      this.queue.add(() => this.run())
     })
   }
 
   // Run the algorithm
   async run () {
+    if (!this.shouldHandleIncoming()) return
```

Comment:
Was this added as a result of the queue logic?

Reply:
yes, suppose that the peer in the first run iteration get the maxPeers connections that can have. (e.g. 4) and a some point two connections are closed, so we will have 2 jobs in the queue to execute `run`.

Now if in the first `run` you get 2 new connections, the second `run` doesn't have to execute anything we are already full of capacity.